### ISSUE TITLE
  onAccessTokenExpired() not called if access token is expired

### ISSUE THREAD

**Issue by bonesoul** [blizzapi/blizzapi#333]
https://blizzapi.lukem.net/docs/usage/custom-access-token.html#parameters says that if we enable   validateAccessTokenOnEachQuery,
the blizzapi should be calling

` onAccessTokenExpired: () => { /* do something */ },`

but from my tests, even if i see expired token in api response

`{ error: 'access_token_invalid' }`

the onAccessTokenExpired event is not called.

```
      this.api_clients[region.code] = new BlizzAPI({
        region: region.code as RegionName,
        accessToken: access_token,
        validateAccessTokenOnEachQuery: true,
        refreshExpiredAccessToken: false,
        onAccessTokenExpired: () => {
          console.dir("onAccessTokenExpired");
        },
      });
```

any ideas?

**Comment by lwojcik** [comment]
Hi @bonesoul! 

Thanks for reporting. I'll check it and get back to you.

**Comment by lwojcik** [comment]
Okay, I'm triaging this. I suspect the library might use a wrong condition for declaring access token as invalid / expired (either nobody noticed before or Battle.net API no longer communicates it the way it used to do?). More updates coming soon.

**Comment by lwojcik** [comment]
Okay, I got this.

When custom access token is provided and `validateAccessTokenOnEachQuery` is passed, the library doesn't do anything except returning the error object. **I'm gonna fix this in the next release** so that `refreshExpiredAccessToken` property can be enabled and _actually used in that case_ (as long as there's client id and client secret available to obtain new access token) + `onAccessTokenExpired()` will be called if available. 

To be honest, I hate how the past version of me designed this custom access token flow back in late 2010s. But I also don't want to break the existing behavior quite a few other projects rely on so that fix must do for now. Personally I no longer run any project that consumes Battle.net data so I might be lagging to catch and fix any potential regressions.

**Comment by lwojcik** [comment]
Released in v2.6.0.
